src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/blob.py

```diff
@@ -124,3 +124,143 @@ def _copy_single_blob(source_blob):
+                              destination_container_name=None, blob_type=None,
+                              content_settings=None, metadata=None, validate_content=False,
+                              maxsize_condition=None, max_connections=2, lease_id=None,
+                              if_modified_since=None, if_unmodified_since=None, if_match=None,
+                              if_none_match=None, timeout=None, dryrun=False):
+    """
+    Upload files to storage container as blobs
+
+    :param str source:
+        The directory where the files to be uploaded.
+
+    :param str destination:
+        The string represents the destination of this upload operation. The source can be the
+        container URL or the container name. When the source is the container URL, the storage
+        account name will parsed from the URL.
+
+    :param str pattern:
+        The pattern is used for files globing. The supported patterns are '*', '?', '[seq]',
+        and '[!seq]'.
+
+    :param bool dryrun:
+        Show the summary of the operations to be taken instead of actually upload the file(s)
+    """
+    def _append_blob(file_path, blob_name):
+        if not client.exists(destination_container_name, blob_name):
+            client.create_blob(
+                container_name=destination_container_name,
+                blob_name=blob_name,
+                content_settings=content_settings,
+                metadata=metadata,
+                lease_id=lease_id,
+                if_modified_since=if_modified_since,
+                if_match=if_match,
+                if_none_match=if_none_match,
+                timeout=timeout)
+
+        return client.append_blob_from_path(
+            container_name=destination_container_name,
+            blob_name=blob_name,
+            file_path=file_path,
+            progress_callback=lambda c, t: None,
+            validate_content=validate_content,
+            maxsize_condition=maxsize_condition,
+            lease_id=lease_id,
+            timeout=timeout)
+
+    def _upload_blob(file_path, blob_name):
+        return client.create_blob_from_path(
+            container_name=destination_container_name,
+            blob_name=blob_name,
+            file_path=file_path,
+            progress_callback=lambda c, t: None,
+            content_settings=content_settings,
+            metadata=metadata,
+            validate_content=validate_content,
+            max_connections=max_connections,
+            lease_id=lease_id,
+            if_modified_since=if_modified_since,
+            if_unmodified_since=if_unmodified_since,
+            if_match=if_match,
+            if_none_match=if_none_match,
+            timeout=timeout)
+
+    upload_action = _upload_blob if blob_type == 'block' or blob_type == 'page' else _append_blob
+
+    if dryrun:
+        print('upload action: from {} to {}'.format(source, destination))
```

Comment:
discussed with @derekbekoe offline, print to standard output is fine here since the command itself doesn't return anything.